graphql_compiler/compiler/emit_cypher.py

```diff
@@ -127,10 +148,64 @@ def _emit_with_clause_components(cypher_steps):
     return result
 
 
+def _emit_with_clause_components_for_current_fold_scope(current_fold_scope_cyphersteps):
+    """Emit the component strings for Cypher WITH clause for current fold."""
+    # At this point we should already have emitted all component strings from previous CypherSteps,
+    # whether or not they were in fold scopes or not.
+    result = []
+
+    vertex_names = {}
+    for cypher_step in current_fold_scope_cyphersteps:
+        fold_scope_location = cypher_step.as_block.location
+        full_vertex_name = fold_scope_location_full_path_name(fold_scope_location)
+        vertex_names[u'collect(' + full_vertex_name + ')'] = u'collected_' + full_vertex_name
+
+    # Sort the locations, to ensure a deterministic order.
+    for index, collect_call in enumerate(sorted(vertex_names)):
+        # haha get it? collect_call because we're calling the function collect()?
```

Comment:
I don't think this comment is contributing to the code clarity or maintainability, I'd remove it.